add examples from JS the goodstuff talk

ttalhouk opened the pull request:
@kelsonic 

Here are some samples from the JS goodstuff talk on Frontend Masters.  It has examples of passing functions around as arguments, methods, closure, curry functions etc.

ttalhouk committed:
add examples from JS the goodstuff talk

kelsonic commented:
This looks great! One quick fix before merging:

Please cleanup the console.logs at the bottom of the file, or at least have comments underneath them stating what they output.

Thanks Talal! :)

ttalhouk committed:
find missing number in a sequence solutions

ttalhouk committed:
add comments to the console logs for running functions

ttalhouk commented:
@kelsonic 
The console logs were for running the functions above to see what they do.  I added some notes to it.  Also added a missing number algorithm solutions 3 ways.
